src/progressbar/progressbar.ts

```diff
@@ -28,7 +38,17 @@ export class NgbProgressbar {
 
   @Input() type: string;
 
-  @Input() value: number;
+  @Input()
+  set value(value: number) {
+    if (value < 0) {
```

Comment:
You can do this in one-liner, similar to what is done in pagination: https://github.com/ng-bootstrap/core/blob/9c6f488ae75a22519c38299d369a616f22203b77/src/pagination/pagination.ts#L68